internal/manifests/gateway.go

```diff
@@ -0,0 +1,266 @@
+					{
+						Name:      "rbac",
+						ReadOnly:  true,
+						MountPath: path.Join(gateway.LokiGatewayMountDir, gateway.LokiGatewayRbacFileName),
+						SubPath:   "rbac.yaml",
+					},
+					{
+						Name:      "tenants",
+						ReadOnly:  true,
+						MountPath: path.Join(gateway.LokiGatewayMountDir, gateway.LokiGatewayTenantFileName),
+						SubPath:   "tenants.yaml",
+					},
+				},
+				LivenessProbe: &corev1.Probe{
+					Handler: corev1.Handler{
+						HTTPGet: &corev1.HTTPGetAction{
+							Path:   "/live",
+							Port:   intstr.FromInt(8081),
+							Scheme: corev1.URISchemeHTTP,
+						},
+					},
+					TimeoutSeconds:   2,
+					PeriodSeconds:    30,
+					FailureThreshold: 10,
+				},
+				ReadinessProbe: &corev1.Probe{
+					Handler: corev1.Handler{
+						HTTPGet: &corev1.HTTPGetAction{
+							Path:   "/ready",
+							Port:   intstr.FromInt(8081),
+							Scheme: corev1.URISchemeHTTP,
+						},
+					},
+					TimeoutSeconds:   1,
+					PeriodSeconds:    5,
+					FailureThreshold: 12,
+				},
+			},
+		},
+	}
+
+	l := ComponentLabels(LabelLokiStackGatewayComponent, opts.Name)
+	a := commonAnnotations(opts.ConfigSHA1)
+
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   LokiStackGatewayName(opts.Name),
+			Labels: l,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels.Merge(l, GossipLabels()),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        LokiStackGatewayName(opts.Name),
+					Labels:      labels.Merge(l, GossipLabels()),
+					Annotations: a,
+				},
+				Spec: podSpec,
+			},
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RollingUpdateDeploymentStrategyType,
+			},
+		},
+	}
+}
+
+// GatewayConfigMap creates a configMap for rbac.yaml and tenants.yaml
+func GatewayConfigMap(opt Options) (*corev1.ConfigMap, string, error) {
+	cfg := GatewayConfigOptions(opt)
+	rbacConfig, tenantsConfig, err := gateway.Build(cfg)
+	if err != nil {
+		return nil, "", err
+	}
+
+	s := sha1.New()
+	_, err = s.Write(rbacConfig)
+	if err != nil {
+		return nil, "", err
+	}
+	sha1C := fmt.Sprintf("%x", s.Sum(nil))
+
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   LabelLokiStackGatewayComponent,
+			Labels: commonLabels(opt.Name),
+		},
+		BinaryData: map[string][]byte{
+			gateway.LokiGatewayRbacFileName:   rbacConfig,
+			gateway.LokiGatewayTenantFileName: tenantsConfig,
+		},
+	}, sha1C, nil
+}
+
+// GatewayConfigOptions converts Options to gateway.Options
+func GatewayConfigOptions(opt Options) gateway.Options {
+	return gateway.Options{
+		Stack:     opt.Stack,
+		Namespace: opt.Namespace,
+		Name:      opt.Name,
+	}
+}
```

Comment:
nit. Move this for better code organization into the options file maybe?

Reply:
I guess it will be good to keep it here. It will create unnecessary import cycles if I move it to option file.